src/Adapter.ts

```diff
@@ -0,0 +1,11 @@
+import { BrokerUtils } from './types';
+
+export interface AdapterConfig {
```

Comment:
It's more like each adapter has its own `Config` tailored to the needs of the adapter, i.e., `HttpAdapterConfig`, `MqttAdapterConfig`. But I guess this works pretty well for now?

Reply:
True. Maybe adapters would even be able to read their own configurations in the future.